internal/keystore/azure/key-vault-error.go

```diff
@@ -0,0 +1,41 @@
+package azure
+
+import (
+	"net/http"
+	"reflect"
+)
+
+// ResponseError is a wrapper around an error response from the Azure Key Vault service.
+type ResponseError struct {
+	// ErrorCode is the error code returned by the resource provider if available.
+	ErrorCode string
+
+	// StatusCode is the HTTP status code as defined in https://pkg.go.dev/net/http#pkg-constants.
+	StatusCode int
+
+	// RawResponse is the underlying HTTP response.
+	RawResponse *http.Response
+
+	errorResponse errorResponse
+}
+
+// TransportErrToResponseError converts a transport error to a ResponseError.
+func TransportErrToResponseError(terr error) (*ResponseError, bool) {
```

Comment:
Same, does this need to be exported?
Also, what exactly is a transport error and what azure API error type are we handling here?